pulsar-client-cpp/lib/ProducerImpl.cc

```diff
@@ -268,6 +264,14 @@ void ProducerImpl::sendAsync(const Message& msg, SendCallback callback) {
         // If batching is enabled we compress all the payloads together before sending the batch
         payload = CompressionCodecProvider::getCodec(conf_.getCompressionType()).encode(payload);
     }
+    uint32_t compressedSize = payload.readableBytes();
+    if (compressedSize > Commands::MaxMessageSize) {
+        if (conf_.getBlockIfQueueFull()) {
```

Comment:
oops. ambiguous.

it's the getBlockIfQueueFull stuff that doesn't make any sense to me.  at the least it deserves a comment because it's bizarre a too large message  would have anything to do with the send queue size.